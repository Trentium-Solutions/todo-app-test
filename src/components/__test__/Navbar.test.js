import React from 'react'
import {render, screen} from '@testing-library/react'
import {createRoot} from 'react-dom/client'
import Navbar from '../Navbar'
import {Provider} from 'react-redux'
import {PersistGate} from 'redux-persist/integration/react'
import {store, persistore} from '../../state/store'
import TestRenderer from 'react-test-renderer'

describe('navbar test suite', () => {
  const navbarComponent = (
    <Provider store={store}>
      <PersistGate persistor={persistore}>
        <Navbar />
      </PersistGate>
    </Provider>
  )
  it('renders without crashing', () => {
    const div = document.createElement('div')
    const root = createRoot(div)
    root.render(<Navbar />)
  })
  it('renders buttons correctly', () => {
    render(navbarComponent)
    expect(screen.getAllByText('Todolist')).toBeTruthy()
  })
  it('matches snapshot navbar', () => {
    const tree = TestRenderer.create(navbarComponent).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
