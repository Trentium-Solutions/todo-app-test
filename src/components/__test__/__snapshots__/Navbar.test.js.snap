// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`navbar test suite matches snapshot navbar 1`] = `
<nav
  className="navbar navbar-dark navbar-expand-lg bg-dark navbar-fixed"
>
  <div
    className="container-fluid"
  >
    <a
      className="navbar-brand fs-4"
      href="/"
    >
      Todolist
    </a>
    <button
      aria-controls="navbarSupportedContent"
      aria-expanded="false"
      aria-label="Toggle navigation"
      className="navbar-toggler"
      data-bs-target="#navbarSupportedContent"
      data-bs-toggle="collapse"
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      className="collapse navbar-collapse"
      id="navbarSupportedContent"
    >
      <ul
        className="navbar-nav me-auto mb-2 mb-lg-0 d-flex justify-content-end"
      />
      <div
        className="cursor hover"
        onClick={[Function]}
        title="Change Theme"
      >
        <label
          className="text-light me-1 cursor margin"
          htmlFor="theme"
        >
          Light 
           Mode
        </label>
        <i
          className="fa-solid fa-sun text-light fs-5 me-3"
          id="theme"
        />
      </div>
      <span
        className="text-light fs-6 me-2"
      >
        Stay organized & focused.
      </span>
    </div>
  </div>
</nav>
`;
